applications/services/power/power_service/power.c

```diff
@@ -34,6 +34,63 @@ static ViewPort* power_battery_view_port_alloc(Power* power) {
     return battery_view_port;
 }
 
+static void power_start_auto_shutdown_timer(Power* power) {
+    furi_timer_start(power->auto_shutdown_timer, furi_ms_to_ticks(power->shutdown_idle_delay_ms));
+}
+
+static void power_stop_auto_shutdown_timer(Power* power) {
+    furi_timer_stop(power->auto_shutdown_timer);
+}
+
+static void power_input_event_callback(const void* value, void* context) {
+    furi_assert(value);
+    furi_assert(context);
+    const InputEvent* event = value;
+    Power* power = context;
+    if(event->type == InputTypePress) {
+        power_start_auto_shutdown_timer(power);
+    }
+}
+
+static void power_auto_shutdown_arm(Power* power) {
+    if(power->shutdown_idle_delay_ms) {
+        power->input_events_subscription =
+            furi_pubsub_subscribe(power->input_events_pubsub, power_input_event_callback, power);
+        power_start_auto_shutdown_timer(power);
+    }
+}
+
+static void power_auto_shutdown_inhibit(Power* power) {
+    power_stop_auto_shutdown_timer(power);
+    if(power->input_events_subscription) {
+        furi_pubsub_unsubscribe(power->input_events_pubsub, power->input_events_subscription);
+        power->input_events_subscription = NULL;
+    }
+}
+
+static void power_auto_shutdown_timer_callback(void* context) {
+    furi_assert(context);
+    Power* power = context;
+    power_auto_shutdown_inhibit(power);
+    power_off(power);
+}
+
+static void auto_shutdown_update(Power* power) {
+    uint32_t old_time = power->shutdown_idle_delay_ms;
+    LOAD_POWER_SETTINGS(&power->shutdown_idle_delay_ms);
```

Comment:
Not good.

Reply:
Sorry, my job hasn't left me a lot of free time lately. I understand the issue and I'm working on it

Reply:
issue solved with commit 82f23547278080bfd80fcea1552960fb81fd0cf5